aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1465,4 +1465,30 @@ Tensor& diag_out(Tensor &result, const Tensor& self, int64_t dimension) {
   return result;
 }
 
+Tensor trace(const Tensor& self) {
+  Tensor result = at::empty({}, self.options());
+  AT_DISPATCH_ALL_TYPES(self.scalar_type(), "trace", [&] {
+    scalar_t sum = 0;
+    const auto* t_data = self.data_ptr<scalar_t>();
+
+    int64_t i = 0;
+    int64_t t_stride_0, t_stride_1, t_diag_size;
+
+    TORCH_CHECK(self.dim() == 2, "trace: expected a matrix, but got tensor with dim ", self.dim());
+
+    t_stride_0 = self.stride(0);
+    t_stride_1 = self.stride(1);
+
+    t_diag_size = std::min(self.size(0), self.size(1));
+    while(i < t_diag_size) {
```

Comment:
nit: for loop looks nicer here, we save 2 lines of code

```
for (int64_t i = 0; i < t_diag_size, i++) {
...
}
```

Reply:
done